main.go

```diff
@@ -649,47 +484,52 @@ func (c *config) structSelection(file ast.Node) (int, int, error) {
 	}
 
 	// if field name has been specified as well, only select the given field
-	if c.fieldName != "" {
-		return c.fieldSelection(encStruct)
+	if cfg.fieldName != "" {
+		return cfg.fieldSelection(encStruct)
 	}
 
-	start := c.fset.Position(encStruct.Pos()).Line
```

Comment:
(preexisting, but these should probably be fset.PositionFor(pos, false). It's a rare gotcha that fset.Position will consider //line directives and actually point at a different file).

Just wanted to make this comment to share knowledge. Let's leave it as-is.